product_packaging_type_required/models/product_packaging_type.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2019 Camptocamp (<http://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+
+
+class ProductPackagingType(models.Model):
+    _inherit = "product.packaging.type"
+
+    required = fields.Boolean()
+
+    @api.model
+    def cron_check_create_required_packaging(self, limit=0):
+        """if limit=0, the method will not apply a limit to process missing
+        packages.
+        """
+        existing_products = self.env["product.product"].search(
+            [("type", "in", ("product", "consu"))]
+        )
+        i = 0
+        required_packaging_types = self.search([("required", "=", True)])
+        for product in existing_products:
+            if limit and i == limit:
+                break
```

Comment:
this is wrong in the original implementation. The limit is never used in fact.
The minimum correction would be:

```suggestion
        i = 0
        required_packaging_types = self.search([("required", "=", True)])
        for i, product in enumerate(existing_products):
            if limit and i == limit:
                break
```
but to be honest I don't find this very effective because you will  always process the same products and packaging.
IMO the only reason why it worked so far is that the limit was never used.

**Hence, please do not fix it.** 
Add a `FIXME: limit is never used and the whole method should be refactored`.

There are no tests too, so it's better to refactor it without them.
I'll have a look at this.

Reply:
Thanks @simahawk for spotting this. PR has been updated per your request.